core/web/nodes_controller.go

```diff
@@ -55,7 +74,14 @@ func (n *nodesController[R]) Index(c *gin.Context, size, page, offset int) {
 		nodes, count, err = n.nodeSet.NodeStatuses(c, offset, size)
 	} else {
 		// fetch nodes for chain ID
-		nodes, count, err = n.nodeSet.NodeStatuses(c, offset, size, id)
+		// backward compatibility
+		rid := &relay.ID{}
```

Comment:
Can you use a value instead of a pointer here to avoid the deference on 84?
```suggestion
		var rid relay.ID
```
`UnmarshalString` should still be auto-upgraded :thinking:

Reply:
nice. i always forget about the auto upgrade. done